Assets/LeapMotionModules/Interaction/Scripts/InteractionController.cs

```diff
@@ -0,0 +1,1382 @@
+﻿/******************************************************************************
+ * Copyright (C) Leap Motion, Inc. 2011-2017.                                 *
+ * Leap Motion proprietary and  confidential.                                 *
+ *                                                                            *
+ * Use subject to the terms of the Leap Motion SDK Agreement available at     *
+ * https://developer.leapmotion.com/sdk_agreement, or another agreement       *
+ * between Leap Motion and you, your company or other organization.           *
+ ******************************************************************************/
+
+using InteractionEngineUtility;
+using Leap.Unity.Space;
+using Leap.Unity.Interaction.Internal;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Leap.Unity.RuntimeGizmos;
+using Leap.Unity.Attributes;
+
+namespace Leap.Unity.Interaction {
+
+  /// <summary>
+  /// Specified on a per-object basis to allow Interaction objects
+  /// to ignore hover for the left hand, right hand, or both hands.
+  /// </summary>
+  public enum IgnoreHoverMode { None, Left, Right, Both }
+
+  /// <summary>
+  /// The Interaction Engine can be controlled by hands tracked by the
+  /// Leap Motion Controller, or by remote-style held controllers
+  /// such as the Oculus Touch or Vive controller.
+  /// </summary>
+  public enum ControllerType { Hand, VRController }
+
+  public abstract class InteractionController : MonoBehaviour {
```

Comment:
[DisallowMultipleComponent](https://docs.unity3d.com/ScriptReference/DisallowMultipleComponent.html) would probably be great for this class.